.github/workflows/task-test.yml

```diff
@@ -51,7 +51,32 @@ jobs:
         id: mode
         run: |
           [[ -z "${{ inputs.container }}" ]] && echo "mode=sudo" >> $GITHUB_OUTPUT || echo "mode=" >> $GITHUB_OUTPUT
-      - name: Deps checkout
+      - name: Check if node20 is Supported
```

Comment:
since duplicated logic, I think we should have a dedicated action/file for this.

Reply:
We can't have it on a different workflow because we cannot call another workflow from a step.
We can implement our own wrapper for this action (and others that also have duplicate variants) in JavaScript. But this is much more complicated.
splitting the entire workflow is possible, but requires moving the decision logic to all the workflow that calls this one, and maintaining two workflows for testing (most of the logic is still shared - how to build, what to test, when and how to upload)